### PR TITLE
Adds OCIO scraping to Nuke. Extracts OCIO scraping from Maya utils in…

### PR DESCRIPTION
**Pull request by jesseconductor**
…to it's own file.

This is to address the issue that ayce-render was experiencing.

### CODE REVIEW

File: conductor/lib/nuke_utils.py [hunk 1]
```diff
@@ -107,6 +134,9 @@ def collect_dependencies(write_nodes, views, dependency_knobs=None):
     '''
     dependency_knobs = dependency_knobs or {}
     deps = set()
+    
+    # A custom OCIO config is defined on the root node
+    deps = deps.union(set(scrape_ocio_dependencies()))
```
Comment (hoolymama) [inline conductor/lib/nuke_utils.py:139]: Since `deps` is an empty **set**, is there any need to `union`?

File: conductor/lib/nuke_utils.py [hunk 2]
```diff
@@ -84,6 +84,33 @@ def get_plugins_info():
...
+        return paths
+    
+    else:
+        return []
+
+def get_ocio_config_filepath():
+    '''
+    Return the OCIO config filepath from Nuke's root node.
+
+    Only return the filepath if color managment is set to use a custom OCIO
```
Comment (flebel) [inline conductor/lib/nuke_utils.py:104]: `s/managment/management/`

File: conductor/lib/ocio_utils.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    search_path_str = _get_ocio_search_path(config_filepath)
+    logger.warning("Could not find PyOpenColorIO library.  Resorting to basic yaml loading")
+    config_dirpath = os.path.dirname(config_filepath)
+    search_paths = search_path_str.split(os.pathsep)
+    logger.debug("Resolving config seach paths: %s", search_paths)
+    for path in search_paths:
+        # If the path is relative, resolve it
+        if not os.path.isabs(path):
+            path = os.path.join(config_dirpath, path)
+            logging.debug("Resolved relative path '%s' to '%s'", )
```
Comment (flebel) [inline conductor/lib/ocio_utils.py:52]: Missing arguments.

**Review by flebel — APPROVED** [review]
+0

File: conductor/lib/nuke_utils.py [hunk 4]
```diff
@@ -84,6 +84,33 @@ def get_plugins_info():
...
+    config.
+    '''
+    
+    ocio_config_path = None
+
+    if nuke.toNode("root").knobs()["OCIO_config"].value() == "custom":
```
Comment (hoolymama) [inline conductor/lib/nuke_utils.py:110]: NIT: A stylistic thing I read somewhere - "Try to get out of the function early." 
If applied here, you would test for `not custom` and return there. Then you don't need to indent long lines. I think it's also a good habit when you have several tests: 
```
if not a:
   return None
if b:
   return None
return something
```

File: conductor/lib/ocio_utils.py (hunk 3 above)
Comment (hoolymama) [inline conductor/lib/ocio_utils.py]: CHANGE: Are values missing after the comma?

File: conductor/lib/nuke_utils.py (hunk 2 above)
Comment (jesseconductor) [reply conductor/lib/nuke_utils.py:104]: fixed

File: conductor/lib/ocio_utils.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+        if not os.path.isdir(path):
+            logger.warning("OCIO search path does not exist: %s", path)
+            continue
```
Comment (hoolymama) [inline conductor/lib/ocio_utils.py:57]: NIT: In a similar vein to getting out of a function early,  you can put opportunities to continue high up in the loop. It allows people reading the code to stop thinking about things earlier.

File: conductor/lib/ocio_utils.py (hunk 3 above)
Comment (jesseconductor) [reply conductor/lib/ocio_utils.py]: fixed

**Review by hoolymama — CHANGES_REQUESTED** [review]
request changes, only because of the string substitution thing - missing stuff after the comma. otherwise looks good.

File: conductor/lib/nuke_utils.py (hunk 1 above)
Comment (jesseconductor) [reply conductor/lib/nuke_utils.py:139]: good point

**Review by hoolymama — APPROVED** [review]

File: conductor/lib/nuke_utils.py (hunk 4 above)
Comment (jesseconductor) [reply conductor/lib/nuke_utils.py:110]: That's true - I think it makes a lot of sense where the rest of the function has a lot more to it, though I'm also not a fan of multiple return statements and I think you have to choose which to do it based on the specific case.

File: conductor/lib/ocio_utils.py (hunk 5 above)
Comment (jesseconductor) [reply conductor/lib/ocio_utils.py:57]: This was a copy/paste. I think there is a lot of room for improvements. In this case, `continue` can be dependent on the conditional immediately above it.

**Pull request merged**